### PR TITLE
Add a git sample app

### PR DESCRIPTION
**Pull request by ForestEckhardt**
- The sample demonstrates both the base functionality as well as how to
configure git credentials and provides user a way to test their binding
by allowing them to clone a private repository.

Quick note on testing this. I have put in the address for the Git buildpack at `paketo-buildpacks` as opposed to `paketo-community` to try and reduce the friction after the promotion. So when testing please bear that in mind.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: git/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# Git Sample Application
+
+## Perpare Application
+
+The Git buildpack requires that there be a valid `.git` folder in the
+application source directory. In order to provide and example application, the
```
Comment (dmikusa-pivotal) [inline git/README.md:6]:
```suggestion
application source directory. In order to provide an example application, the
```

File: git/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+```bash
+pack build git-sample --path app \
+  --buildpack gcr.io/paketo-buildpacks/git
+```
+
+### Viewing
+
+The Git buildpack set the `REVISION` environment variable during build for
+subsequent buildpacks to take advantage of during their build processes. It
+also sets this same value as a lable on the image. This value can be seen with
```
Comment (dmikusa-pivotal) [inline git/README.md:27]:
```suggestion
also sets this same value as a label on the image. This value can be seen with
```

File: git/app/.git.bak/.git.bak/config [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+	repositoryformatversion = 0
+	filemode = true
+	bare = false
+	logallrefupdates = true
+	ignorecase = true
+	precomposeunicode = true
+[duet "env"]
+	git-author-initials = rm
+	git-author-name = Ryan Moran
+	git-author-email = rmoran@vmware.com
```
Comment (ryanmoran) [inline git/app/.git.bak/.git.bak/config:11]: Hmm. Maybe don't check my name and email address into the codebase?

**Review by ryanmoran — APPROVED** [review]

**Comment by ForestEckhardt** [comment]
@ryanmoran There are not currently any smoke test for this sample. The current workflow that exists does not really support standalone buildpack tests they are typically running tests against builders. There is currently an issue to add test to the ca-certificates samples and because the git sample falls into a similar category I omitted the tests for the time being.

Is this a deal breaker for you in approving this sample?

**Comment by ryanmoran** [comment]
The point of the smoke tests is to confirm that these samples work. If we want a sample for this buildpack, then we should include a test so that we can make sure our samples actually work for users.

**Comment by ForestEckhardt** [comment]
Alright I will try and set up a paradigm that works for single buildpack tests. I think that the test will be relatively straight forward!

File: git/git_test.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	"github.com/paketo-buildpacks/samples/tests"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	. "github.com/onsi/gomega"
+	. "github.com/paketo-buildpacks/occam/matchers"
+)
+
+var builders tests.BuilderFlags
+var suite spec.Suite
```
Comment (ryanmoran) [inline git/git_test.go:21]:
```suggestion
```

Do we need this? Looks like it shadows the `suite` variable that gets defined in the `TestNginx` method.

File: git/git_test.go [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+func init() {
+	flag.Var(&builders, "name", "the name a builder to test with")
+}
+
+func TestNginx(t *testing.T) {
```
Comment (ryanmoran) [inline git/git_test.go:27]:
```suggestion
func TestGit(t *testing.T) {
```

**Review by ryanmoran — APPROVED** [review]

**Pull request merged**